x/observer/keeper/msg_server_add_block_header.go

```diff
@@ -52,28 +83,25 @@ func (k msgServer) AddBlockHeader(goCtx context.Context, msg *types.MsgAddBlockH
 	/**
 	 * Vote finalized, add block header to store
 	 */
-	_, found = k.GetBlockHeader(ctx, msg.BlockHash)
-	if found {
-		hashString, err := common.HashToString(msg.ChainId, msg.BlockHash)
-		if err != nil {
-			return nil, cosmoserrors.Wrap(err, "block hash conversion failed")
-		}
-		return nil, cosmoserrors.Wrap(types.ErrBlockAlreadyExist, hashString)
-	}
-
-	// Check timestamp
-	err = msg.Header.ValidateTimestamp(ctx.BlockTime())
-	if err != nil {
-		return nil, cosmoserrors.Wrap(types.ErrInvalidTimestamp, err.Error())
-	}
-
-	// NOTE: error is checked in BasicValidation in msg; check again for extra caution
-	pHash, err := msg.Header.ParentHash()
-	if err != nil {
-		return nil, cosmoserrors.Wrap(types.ErrNoParentHash, err.Error())
-	}
 
 	// TODO: add check for parent block header's existence here https://github.com/zeta-chain/node/issues/1133
+	if !found {
```

Comment:
`found` is defined above in the method, I think we should provide a more explicit name to make the code easier to understand

Reply:
done at the expense of slightly more read ops for the sake of readability. 
https://github.com/zeta-chain/node/pull/1374/commits/be1b98d466cda26bdadcc982fc7987f7d05ff642